components/DocumentModels.vue

```diff
@@ -0,0 +1,149 @@
+<script setup lang="ts">
+import type { IAllDocumentModels, TrackModel } from "@bcc-code/bmm-sdk-fetch";
+import type { IDiscoverableGroup } from "~/composables/discover";
+import { breakpointsTailwind, useBreakpoints } from "@vueuse/core";
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  models: IAllDocumentModels[];
+}>();
+
+const convertModels = (models: IAllDocumentModels[]) => {
+  let currentSection: IDiscoverableGroup["items"] = [];
+  const result: IDiscoverableGroup[] = [];
+  models.forEach((el, i) => {
+    if (el.type === "section_header") {
+      currentSection = [];
+      result.push({
+        header: el,
+        items: currentSection,
+      });
+    } else if (i === 0) {
+      currentSection = [el];
+      result.push({
+        header: null,
+        items: currentSection,
+      });
+    } else {
+      currentSection.push(el);
+    }
+  });
+  return result;
+};
+
+const { setQueue } = useNuxtApp().$mediaPlayer;
+
+const playItem = (item: TrackModel, group: IDiscoverableGroup) => {
+  const items = group.items.filter((c): c is TrackModel => c.type === "track");
+  setQueue(
+    items,
+    items.findIndex((track) => track.id === item.id),
+  );
+};
+
+// We remove the hostname so that we use SPA links (without full page refresh)
+const parseLink = (link: string) => {
+  const url = new URL(link);
```

Comment:
Are all URLs absolute by definition? If the URL is not absolute, this line will throw an error `Uncaught TypeError: Failed to construct 'URL': Invalid URL`.

Reply:
yes. They are absolute.